### ISSUE TITLE
Security logging profile for bot defense 

### ISSUE THREAD

**Issue by yossi-r** [F5Networks/f5-appsvcs-extension#66]
##### Do you already have an issue opened with F5 support?
Github Issues are consistently monitored by F5 staff, but should be considered as best effort only and you should not expect to receive the same level of response as provided by F5 Support. Please [open an case](https://support.f5.com/csp/article/K2633) with F5 if this is a critical issue.

##### ISSUE TYPE
 - Bug Report
 - Feature Idea

##### AS3 BUILD/ VERSION

-->
```
3.7
```

##### BIGIP VERSION

```
13.1.1
```

##### SUMMARY

```
i  need to create a security logging profile with remote logging for AWAF and bots. 
the current version can only create a log destination of type:
remote-syslog
remote-high-speed-log

while bigip expects the logging profile for dos/bots to be of type splunk. 


```

##### STEPS TO REPRODUCE

```
{
    "class": "AS3",
    "action": "deploy",
    "persist": true,
    "declaration": {
        "class": "ADC",
        "schemaVersion": "3.7.0",
        "id": "urn:uuid:33045210-3ab8-4636-9b2a-c98d22ab915d",
        "label": "shared profiles and objects",
        "Common": {
            "class": "Tenant",
            "Shared": {
                "class": "Application",
                "template": "shared",
                "TEST_Security_Log_Profile_49": {
                "class": "Security_Log_Profile",
                "remark": "description",
                "dosApplication": {
                    "localPublisher": {
                        "bigip": "/Common/local-db-publisher"
                    },
                    "remotePublisher": {
                        "use": "logPub"
                    }
                },
                "botDefense": {
                    "localPublisher": {
                        "bigip": "/Common/local-db-publisher"
                    },
                    "logChallengedRequests": true,
                    "logIllegalRequests": false,
                    "logLegalRequests": true,
                    "logBotSignatureMatchedRequests": true,
                    "logCaptchaChallengedRequests": true
                }
            },
            "logPub": {
                "class": "Log_Publisher",
                "destinations": [
                    {
                        "use": "logDest"
                    }
                ]
            },
            "logDest": {
                "class": "Log_Destination",
                "type": "remote-syslog",
                "remoteHighSpeedLog": {
                    "use": "highSpeed"
                }
            },
            "highSpeed": {
                "class": "Log_Destination",
                "type": "remote-high-speed-log",
                "pool": {
                    "use": "thePool"
                }
            },
            "thePool": {
                "class": "Pool"
            }
        }
    }
            }
}
```

##### EXPECTED RESULTS

```
logging profile with all the dependencies should have been created. 
```

##### ACTUAL RESULTS

```
{
    "results": [
        {
            "message": "declaration failed",
            "response": "0107161f:3: Log publisher '/Common/Shared/logPub' used by Application DoS Security log profile can have only ArcSight or Splunk destinations.",
            "code": 422,
            "host": "localhost",
            "tenant": "Common",
            "runTime": 3745
        },
        {
            "message": "declaration failed",
            "response": "0107161f:3: Log publisher '/Common/Shared/logPub' used by Application DoS Security log profile can have only ArcSight or Splunk destinations.",
            "code": 422,
            "host": "localhost",
            "tenant": "Common",
            "runTime": 3393
        }
    ],
    "declaration": {
        "Common": {
            "class": "Tenant",
            "Shared": {
                "class": "Application",
                "template": "shared",
                "TEST_Security_Log_Profile_49": {
                    "class": "Security_Log_Profile",
                    "remark": "description",
                    "dosApplication": {
                        "localPublisher": {
                            "bigip": "/Common/local-db-publisher"
                        }
                    },
                    "botDefense": {
                        "localPublisher": {
                            "bigip": "/Common/local-db-publisher"
                        },
                        "logChallengedRequests": true,
                        "logIllegalRequests": false,
                        "logLegalRequests": true,
                        "logBotSignatureMatchedRequests": true,
                        "logCaptchaChallengedRequests": true
                    }
                },
                "logPub": {
                    "class": "Log_Publisher",
                    "destinations": [
                        {
                            "use": "logDest"
                        }
                    ]
                },
                "logDest": {
                    "class": "Log_Destination",
                    "type": "remote-syslog",
                    "remoteHighSpeedLog": {
                        "use": "highSpeed"
                    }
                },
                "highSpeed": {
                    "class": "Log_Destination",
                    "type": "remote-high-speed-log",
                    "pool": {
                        "use": "thePool"
                    }
                },
                "thePool": {
                    "class": "Pool"
                }
            }
        },
        "class": "ADC",
        "schemaVersion": "3.7.0",
        "id": "urn:uuid:33045210-3ab8-4636-9b2a-c98d22ab915d",
        "label": "shared profiles and objects",
        "updateMode": "selective",
        "controls": {
            "archiveTimestamp": "2019-02-20T17:55:48.587Z"
        }
    },
    "code": 422
}
```

**Comment by dstokesf5** [comment]
Thank you for your feedback. We are already working on adding support for Splunk log destinations, and it should be available in the 3.10 release.

**Comment by kaustriaf5** [comment]
Tracked internally as 850 and released as part of [3.10](https://github.com/F5Networks/f5-appsvcs-extension/releases/tag/v3.10.0) 
@yossi-r please see the 3.10 release. Closing the issue.
